src/cargo/lib.rs

```diff
@@ -9,53 +9,92 @@
+//!   Sources are uniquely identified by a `SourceId`. Sources are implemented in the [`sources`]
+//!   directory.
+//! - [`util`]:
+//!   This directory contains generally-useful utility modules.
+//! - [`util::config`]:
+//!   This directory contains the config parser. It makes heavy use of
+//!   [serde](https://serde.rs/) to merge and translate config values. The
+//!   [`util::Config`] is usually accessed from the
+//!   [`core::Workspace`]
+//!   though references to it are scattered around for more convenient access.
+//! - [`util::toml`]:
+//!   This directory contains the code for parsing `Cargo.toml` files.
+//!   - [`ops::lockfile`]:
+//!     This is where `Cargo.lock` files are loaded and saved.
 //!
-//! This API documentation is published on each push of rust-lang/cargo master branch.
-//! In other words, it always reflects the latest doc comments in source code on master branch.
+//! Related crates:
+//! - [`cargo-platform`](https://crates.io/crates/cargo-platform)
+//!   ([nightly docs](https://doc.rust-lang.org/nightly/nightly-rustc/cargo_platform)):
+//!   This library handles parsing `cfg` expressions.
+//! - [`cargo-util`](https://crates.io/crates/cargo-util)
+//!   ([nightly docs](https://doc.rust-lang.org/nightly/nightly-rustc/cargo_util)):
+//!   This contains general utility code that is shared between cargo and the testsuite
+//! - [`crates-io`](https://crates.io/crates/crates-io)
+//!   ([nightly docs](https://doc.rust-lang.org/nightly/nightly-rustc/crates_io)):
+//!   This contains code for accessing the crates.io API.
+//! - [`home`](https://crates.io/crates/home):
+//!   This library is shared between cargo and rustup and is used for finding their home directories.
+//!   This is not directly depended upon with a `path` dependency; cargo uses the version from crates.io.
+//!   It is intended to be versioned and published independently of Rust's release system.
+//!   Whenever a change needs to be made, bump the version in Cargo.toml and `cargo publish` it manually, and then update cargo's `Cargo.toml` to depend on the new version.
+//! - [`cargo-test-support`](https://github.com/rust-lang/cargo/tree/master/crates/cargo-test-support)
+//!   ([nightly docs](https://doc.rust-lang.org/nightly/nightly-rustc/cargo_test_support/index.html)):
+//!   This contains a variety of code to support writing tests
+//! - [`cargo-test-macro`](https://github.com/rust-lang/cargo/tree/master/crates/cargo-test-macro)
+//!   ([nightly docs](https://doc.rust-lang.org/nightly/nightly-rustc/cargo_test_macro/index.html)):
+//!   This is the `#[cargo_test]` proc-macro used by the test suite to define tests.
+//! - [`credential`](https://github.com/rust-lang/cargo/tree/master/crates/credential)
+//!   This subdirectory contains several packages for implementing the
+//!   experimental
+//!   [credential-process](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#credential-process)
```

Comment:
The layout looks fine here, though I prefer reference-style links. They are more readable at source code level.

Reply:
I try them every once in a while but always feel like they are more of a pain to initially write (duplicating the referencing, jumping back and forth) and then dealing with later (more places to touch when updating).